pytest_bdd/plugin.py

```diff
@@ -103,6 +103,8 @@ def item_key(item):
             declaration_order = getattr(item.function, "__pytest_bdd_counter__", 0)
         else:
             declaration_order = 0
-        return (item.reportinfo()[:2], declaration_order)
+        func, linenum = item.reportinfo()[:2]
+        info = (func, linenum) if linenum is not None else (func, -1)
```

Comment:
I see the items that don't have any line number are coming from the lint and have PyLintItem type. In this case it is OK if they go first.

But if it is unpacked, let's not pack it back then.
func, linenum = item.reportinfo()[:2]
return (func, linenum if linenum is not None else -1, declaration_order)

Reply:
Thank you for the feedback. I'll try and test your suggestion this weekend and will update CHANGES.rst too.